### PR TITLE
Added cctl

### PR DESCRIPTION
**Pull request by trvv**
Added [cctl](https://github.com/bmofa/cctl) as a submodule to custard. 
Added the installation of cctl to the main `make install` target.

### CODE REVIEW

**Comment by Sweets** [comment]
Good work. I've decided that I'll leave this pull request open, though, until cctl is feature complete. Once that happens it will likely replace custardctl as the main user interface with custard.

**Comment by trvv** [comment]
What should I continue to add? Should I just make the syntax for it custardctl compliant?

**Comment by Sweets** [comment]
Best case-scenario, cctl could become the drop-in replacement for custardctl, without breaking existing configurations. As for feature completeness, at the very least cctl should implement each of the commands that can be output to the custard socket (see the Wiki page or run custardctl commands with `dry` append to the end for socket inputs).

Ideally, it would also implement methods of parsing for specific data (colors with alpha channels in #rrggbbaa format, strings "True" and "False", as well as their lowercase equivalents, evaluating to 1 or 0 respectively, and unsigned integers), as well as helpful error messages that can tell where a user went wrong with an invalid command to cctl.

Parsing for the data can be found at the bottom of src/ipc.c, if need be.

**Comment by trvv** [comment]
Good to know, I'll start working on proper parsing once I get my own setup decently configured. And currently cctl can execute anything that will be sent to the socket, because it just directly sends inputs to the socket with zero parsing (I know, I know).

File: Makefile [hunk 1]
```diff
@@ -37,6 +37,8 @@ $(BUILDPREFIX)/%.o: src/%.c
 install:
 	install -m 755 -D $(BUILDPREFIX)/$(TARGET) $(DESTDIR)$(PREFIX)/bin/$(TARGET)
 	install -m 755 -D contrib/custardctl.py $(DESTDIR)$(PREFIX)/bin/custardctl
+	cd contrib/cctl
+	make install
```
Comment (countingsort) [inline Makefile]: Don't `cd` to call `make` in a sub directory. Use `make -C` instead.

Comment (buffet) [inline Makefile:41]: Don't compile submodules like this. Use `make -C` instead.

**Comment by Snuggle** [comment]
@0xfi Any updates on this?

**Pull request closed without merging**